unity-renderer/Assets/DCLServices/WearablesCatalogService/LambdasWearablesCatalogService.cs

```diff
@@ -0,0 +1,308 @@
+        {
+            foreach (WearableItem wearableItem in wearableItems)
+            {
+                if (WearablesCatalog.ContainsKey(wearableItem.id))
+                    continue;
+
+                wearableItem.SanitizeHidesLists();
+                WearablesCatalog.Add(wearableItem.id, wearableItem);
+
+                if (!wearablesInUseCounters.ContainsKey(wearableItem.id))
+                    wearablesInUseCounters.Add(wearableItem.id, 1);
+            }
+        }
+
+        public void RemoveWearablesFromCatalog(IEnumerable<string> wearableIds)
+        {
+            foreach (string wearableId in wearableIds)
+            {
+                WearablesCatalog.Remove(wearableId);
+                wearablesInUseCounters.Remove(wearableId);
+            }
+        }
+
+        public void RemoveWearablesInUse(IEnumerable<string> wearablesInUseToRemove)
+        {
+            foreach (string wearableToRemove in wearablesInUseToRemove)
+            {
+                if (!wearablesInUseCounters.ContainsKey(wearableToRemove))
+                    continue;
+
+                wearablesInUseCounters[wearableToRemove]--;
+            }
+        }
+
+        public void EmbedWearables(IEnumerable<WearableItem> wearables)
+        {
+            foreach (WearableItem wearableItem in wearables)
+            {
+                WearablesCatalog[wearableItem.id] = wearableItem;
+
+                if (wearablesInUseCounters.ContainsKey(wearableItem.id))
+                    wearablesInUseCounters[wearableItem.id] = 10000; //A high value to ensure they are not removed
+            }
+        }
+
+        public void Clear()
+        {
+            WearablesCatalog.Clear();
+            wearablesInUseCounters.Clear();
+            awaitingWearableTasks.Clear();
+            pendingWearablesToRequest.Clear();
+        }
+
+        UniTask<(WearableResponse response, bool success)> ILambdaServiceConsumer<WearableResponse>.CreateRequest
+            (string endPoint, int pageSize, int pageNumber, CancellationToken cancellationToken) =>
+            lambdasService.Ref.Get<WearableResponse>(
+                "",
+                endPoint,
+                REQUESTS_TIME_OUT_SECONDS,
+                ATTEMPTS_NUMBER,
+                cancellationToken,
+                LambdaPaginatedResponseHelper.GetPageSizeParam(pageSize),
+                LambdaPaginatedResponseHelper.GetPageNumParam(pageNumber));
+
+        private async UniTask<WearableItem> SyncWearablesRequestsAsync(string newWearableId, CancellationToken ct)
+        {
+            if (!awaitingWearableTasks.TryGetValue(newWearableId, out var source))
+            {
+                pendingWearablesToRequest.Add(newWearableId);
+                awaitingWearableTasks[newWearableId] = source = new UniTaskCompletionSource<WearableItem>();
+            }
+
+            await UniTask.DelayFrame(FRAMES_TO_CHECK_FOR_SENDING_PENDING_REQUESTS, cancellationToken: ct);
```

Comment:
It can't work this way: you just skip a frame instead of synchronizing.
I proposed to `Yield(PostLateUpdate)` because `FRAMES_TO_CHECK_FOR_SENDING_PENDING_REQUESTS` = 1

Reply:
Done!